pkg/controllers/dnspolicy/dnspolicy_healthchecks.go

```diff
@@ -101,40 +101,44 @@ func (r *DNSPolicyReconciler) expectedProbesForGateway(ctx context.Context, gw c
 	var healthChecks []*v1alpha1.DNSHealthCheckProbe
 	if dnsPolicy.Spec.HealthCheck == nil {
 		log.V(3).Info("DNS Policy has no defined health check")
-		return nil, nil
+		return healthChecks, nil
 	}
-	ipPattern := `\b(?:\d{1,3}\.){3}\d{1,3}\b`
-	re := regexp.MustCompile(ipPattern)
 
 	for _, address := range gw.Status.Addresses {
 		port := dnsPolicy.Spec.HealthCheck.Port
 		if port == nil {
 			port = &defaultPort
 		}
 
-		matches := re.FindAllString(address.Value, -1)
-		if len(matches) != 1 {
-			log.V(3).Info("Found more or less than 1 ip address")
-			continue
+		matches := strings.Split(address.Value, "/")
+		if len(matches) != 2 {
+			return nil, errors.New(fmt.Sprintf("unable to extract address from address %s ", address.Value))
```

Comment:
Are we happy to error here rather than log. The log seemed quiet if there was to be a problem with the formatting of the adress.Value cc @philbrookes ?

Reply:
I wonder was this a log rather than an error, so that any good values still get processed?

Reply:
I put this back to a log. 👍  removed the return of the error as it's never used.